components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -346,9 +347,22 @@ class GeckoEngine(
             } ?: emptyList()
 
             extensions.forEach { extension ->
+                // https://github.com/mozilla-mobile/android-components/issues/6227 set all installed
+                // extensions to be allowed in private browsing mode. We need to revert back all the
+                // users affected back to default allowPrivateBrowsing, which is false.
+                if (extension.isAllowedInPrivateBrowsing() &&
+                        sharedPref.getBoolean(KEY_DEFAULT_WEB_EXTENSION_ALLOWED_IN_PRIVATE_BROWSING, false)
+                ) {
+                    setAllowedInPrivateBrowsing(extension, false, onSuccess = {})
+                }
+
                 extension.registerActionHandler(webExtensionActionHandler)
                 extension.registerTabHandler(webExtensionTabHandler)
             }
+
+            sharedPref.edit().putBoolean(KEY_DEFAULT_WEB_EXTENSION_ALLOWED_IN_PRIVATE_BROWSING, true)
+                    .apply()
```

Comment:
We shouldn't set this to true every time, only after we reset the default. Otherwise we'd execute this pref write on every start of the app, but we really only need to do it once.